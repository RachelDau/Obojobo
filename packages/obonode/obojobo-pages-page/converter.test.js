jest.mock('obojobo-document-engine/src/scripts/common/index', () => ({
	Registry: {
		getItemForType: type => ({
			slateToObo: () => ({
				slateToOboReturnFor: type
			}),
			oboToSlate: type => ({
				oboToSlateReturnFor: type
			})
		})
	}
}))

jest.mock('obojobo-document-engine/src/scripts/oboeditor/components/node/editor', () => ({
	helpers: {
		slateToObo: child => ({
			componentSlateToOboReturnFor: child
		}),
		oboToSlate: child => ({
			componentOboToSlateReturnFor: child
		})
	}
}))

import Converter from './converter'

describe('Page Converter', () => {
	test('slateToObo converts a Slate node to an OboNode', () => {
		const slateNode = {
			id: 'mockKey',
			type: 'mockType',
			content: {},
			children: [
				{
					type: 'oboeditor.component',
					children: [
						{
							type: 'mockNode'
						}
					]
				}
			]
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			children: [
				{
					type: 'oboeditor.component',
					children: [
						{
							type: 'mockNode'
						}
					]
				}
			],
			content: {
				triggers: 'mock-triggers'
			}
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})
})
