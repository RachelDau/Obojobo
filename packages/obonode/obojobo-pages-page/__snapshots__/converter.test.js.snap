// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Converter oboToSlate converts an OboNode to a Slate node 1`] = `
Object {
  "children": Array [
    Object {
      "componentOboToSlateReturnFor": Object {
        "children": Array [
          Object {
            "type": "mockNode",
          },
        ],
        "type": "oboeditor.component",
      },
    },
  ],
  "content": Object {
    "triggers": "mock-triggers",
  },
  "id": "mockKey",
  "type": "mockType",
}
`;

exports[`Page Converter slateToObo converts a Slate node to an OboNode 1`] = `
Object {
  "children": Array [
    Object {
      "componentSlateToOboReturnFor": Object {
        "children": Array [
          Object {
            "type": "mockNode",
          },
        ],
        "type": "oboeditor.component",
      },
    },
  ],
  "content": Object {},
  "id": "mockKey",
  "type": "mockType",
}
`;
